src/new_storage/storage_manager/handle.rs

```diff
@@ -601,68 +666,135 @@ impl Handles {
     }
 
     /// Gets a shard of the fh.
-    fn get_shard(&self, fh: u64) -> &Arc<RwLock<Vec<FileHandle>>> {
+    fn get_shard(&self, fh: u64) -> &Arc<tokio::sync::RwLock<HashMap<u64, FileHandle>>> {
         let idx = Self::hash(fh);
         self.handles
             .get(idx)
             .unwrap_or_else(|| unreachable!("The array is ensured to be long enough."))
     }
 
     /// Adds a file handle to the collection.
-    pub fn add_handle(&self, fh: FileHandle) {
+    pub async fn add_handle(&self, fh: FileHandle) {
```

Comment:
add_handle has no use

Reply:
Ok, `add_handle` is used in ut, changed to `create_or_open_handle`.